test/helpers/assertions.js

```diff
@@ -140,3 +143,38 @@ function assertContractSaved(expectedAddress, expectedChain, expectedStatus) {
     chai.expect(isExist, "Contract is not saved").to.be.true;
   }
 }
+
+/**
+ * assertContractNotSaved checks that the verification result was not stored on the file system.
+ */
+const assertContractNotSaved = (expectedAddress, expectedChain) => {
+  // Check contract verification result was NOT saved to the disk
+  const fullMatchPath = path.join(
+    config.repository.path,
+    "contracts",
+    "full_match",
+    expectedChain,
+    getAddress(expectedAddress),
+    "metadata.json"
+  )
+  const partialMatchPath = path.join(
+    config.repository.path,
+    "contracts",
+    "partial_match",
+    expectedChain,
+    getAddress(expectedAddress),
+    "metadata.json"
+  )
+  const isExist = fs.existsSync(fullMatchPath) || fs.existsSync(partialMatchPath);
+  chai.expect(isExist, "Contract should not have been saved").to.be.false;
```

Comment:
This might be handy for debugging

```suggestion
  chai.expect(isExist, `Contract should not have been saved, either ${fullMatchPath} | ${partialMatchPath} exists`).to.be.false;  
```

Feel free to include it if you think it helps.

Reply:
will do, thx